content/geometry/CircleCircleArea.h

```diff
@@ -0,0 +1,21 @@
+/**
+ * Author: Takanori MAEHARA, chilli
+ * Date: 2019-11-03
+ * License: CC0
+ * Source: https://github.com/spaghetti-source/algorithm/blob/master/geometry/_geom.cc#L729
+ * Description: Calculates the area of the intersection of 2 circles
+ * Status:
+ */
+
+template<class P>
+double circleCircleArea(P c, double cr, P d, double dr) {
+	if (cr < dr) swap(c, d), swap(cr, dr);
+	auto A = [&](double r, double h) {
+		return r*r*acos(h/r)-h*sqrt(r*r-h*h);
```

Comment:
The removal of the epsilons now means that r < h can happen due to numerical precision, I suspect (though I haven't tested). `h = min(h, r);` may be reasonable? `h < 0` can likely also happen but doesn't seem like a problem.